waspc/packages/deploy/src/providers/fly/createDb/createDb.ts

```diff
@@ -0,0 +1,28 @@
+import { $, question } from 'zx'
+import { exit } from 'process'
+import * as tomlHelpers from '../helpers/tomlFileHelpers.js'
+import { DeploymentInfo } from '../DeploymentInfo.js'
+import { ICreateDbOptions } from './ICreateDbOptions.js'
+import { getInferredBasenameFromServerToml } from '../helpers/tomlFileHelpers.js'
+import { waspSays } from '../helpers/helpers.js'
+
+export async function createDb(region: string, options: ICreateDbOptions) {
+  const tomlFiles = tomlHelpers.getTomlFileInfo(options)
+
+  if (!tomlHelpers.serverTomlExistsInProject(tomlFiles)) {
+    waspSays(`${tomlFiles.serverTomlPath} missing. Skipping server deploy. Perhaps you need to run the "setup" command first?`)
+    exit(1)
+  }
+
+  const inferredBaseName = getInferredBasenameFromServerToml(tomlFiles)
+  const deploymentInfo = new DeploymentInfo(inferredBaseName, region, options, tomlFiles)
+
+  // Creates a DB, waits for it to come up, then links it to the app.
+  // The attachment process shares the DATABASE_URL secret.
+  await $`flyctl postgres create --name ${deploymentInfo.dbName()} --region ${deploymentInfo.region} --vm-size ${options.vmSize} --initial-cluster-size ${options.initialClusterSize} --volume-size ${options.volumeSize}`
```

Comment:
Nice playground! :D Yeah, that is probably more readable to do it the way you suggest in the comment vs. 1 big long string. Will update thnx

Reply:
Because of the way `zx` handles quoting of arrays of string args to commands, I had to change it up slightly, but it still came away looking better I think, thanks!

```ts
  const createArgs = [
    '--name', deploymentInfo.dbName,
    '--region', deploymentInfo.region,
    '--vm-size', options.vmSize,
    '--initial-cluster-size', options.initialClusterSize,
    '--volume-size', options.volumeSize
  ]
  await $`flyctl postgres create ${createArgs}`
```